### PR TITLE
Rearrange document and add API reference for setConsent API.

### PR DESCRIPTION
**Pull request by addb**
## Description
- Rearrange doc into separate section for readability
- Add setConsent API reference with sample

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/api-reference.md [hunk 1]
```diff
@@ -2,22 +2,27 @@
...
+  - [getExperienceCloudId](#getExperienceCloudId)
+  - [resetIdentities](#resetIdentities)
+  - [(optional) setExperienceCloudId](#setExperienceCloudId)
 - Media APIs
-    - [createMediaSession](#createMediaSession)
-    - [sendMediaEvent](#sendMediaEvent)
+  - [createMediaSession](#createMediaSession)
+  - [sendMediaEvent](#sendMediaEvent)
+- Util APIs
+  - [_adb_ISO8601_timestamp](#_adb_ISO8601_timestamp)
```
Comment (praveek) [inline Documentation/api-reference.md:23]: Can you remove the leading underscore and rename this method to `adb_ISO8601_timestamp`?

Comment (addb) [reply Documentation/api-reference.md:23]: Sure, I wanted to also get confirmation that we are okay to make this public that's why did not update it. 
Also how do you think about adding it a ADB_Helper class. 

```
function AEPSdkHelper

  function getISO8601Timestamp
     return _adb_ISO8601_timestamp();
  end function
  
end function 
```

Usage:
```
aepSdkHelper = AEPSdkHelper()
ts = aepSdkHelper. getISO8601Timestamp()
```

Comment (stcheng-adobe) [reply Documentation/api-reference.md:23]: Why do we need to expose this API?

Comment (addb) [reply Documentation/api-reference.md:23]: @stcheng-adobe there is no straight forward way to create ISO timestamp on Roku and the setConsent API will error if customer passes Timestamp in different format. So this helper will make it easy for them to get the ts in right format.

Comment (praveek) [reply Documentation/api-reference.md:23]: I reviewed the PR again, and the term ‘timestamp’ isn’t very clear. I prefer changing the helper’s name to [AEPSDKUtils.getISO8601UTCDateWithMilliseconds](https://github.com/adobe/aepsdk-core-android/blob/main/code/core/src/main/java/com/adobe/marketing/mobile/util/TimeUtils.kt#L90C9-L90C42) to align with our mobile implementation?

File: Documentation/api-reference.md [hunk 2]
```diff
@@ -68,72 +73,187 @@ m.aepSdk = AdobeAEPSDKInit()
...
+```brightscript
+setConsent: function(data as object) as void
+```
+
+- `@param data as object : an associative array that includes consent data to be sent to the edge network.`
+
+##### Example: setConsent using consent object following Adobe 2.0 Standard
+
+> [!Note]
+> The "time" field under "metadata" should contain timestamp in ISO format as shown below.
```
Comment (praveek) [inline Documentation/api-reference.md:230]: should contain ISO 8601 formatted with UTC(Z) time zone, millisecond precision date

File: Documentation/api-reference.md (hunk 1 above)
Comment (addb) [reply Documentation/api-reference.md:23]: @praveek sounds good!

Comment (addb) [reply Documentation/api-reference.md:23]: @praveek @stcheng-adobe 
I was searching the Roku docs again for this and I do see ISO dateTime method. And We can just mention that method to be used for this and not document our internal method.  

```
  dateTime = CreateObject("roDateTime")
  
  isoDate = dateTime.ToISOString("milliseconds")
  isoDateFromSDK = _adb_ISO8601_timestamp()
  
  print("roku method iso date:" + FormatJson(isoDate))
  print("sdk method iso date:" + FormatJson(isoDateFromSDK))
```

Output:
```
roku method iso date:"2024-08-30T03:11:12.395Z"
sdk method iso date:"2024-08-30T03:11:12.395Z"
```

We can also update our method to use this for our logic as well. But I will have to understand why we did not use this in the first place.

File: Documentation/api-reference.md [hunk 3]
```diff
@@ -68,72 +71,189 @@ m.aepSdk = AdobeAEPSDKInit()
...
+}
+
+m.aepSdk.updateConfiguration(configuration)
+```
+
+The `EDGE_CONFIG_ID` value is presented as `Datastream ID` in the [Datastream details](https://experienceleague.adobe.com/docs/experience-platform/edge/datastreams/configure.html?lang=en#view-details) page.
+
+The `EDGE_DOMAIN` value is the first-party domain mapped to the Adobe-provisioned Edge Network domain. For more information, see this [documentation](https://developer.adobe.com/client-sdks/documentation/edge-network/#domain-configuration)
+
+### Configure default consent
```
Comment (praveek) [inline Documentation/api-reference.md:185]: I prefer moving this under the consent section.

File: Documentation/api-reference.md [hunk 4]
```diff
@@ -68,72 +71,189 @@ m.aepSdk = AdobeAEPSDKInit()
...
+```brightscript
+  collectConsentYes = {
+    "consent": [
+      {
+        "standard": "Adobe",
+        "version": "2.0",
+        "value": {
+          "metadata": {
+            ' pass timestamp in ISO format
+            "time" = "2023-10-03T17:23:04.443Z"
```
Comment (praveek) [inline Documentation/api-reference.md:242]: "time" : ...

Comment (praveek) [reply Documentation/api-reference.md:242]: Could you also provide a code snippet for generating the time in this format using the toISOString API?

File: Documentation/api-reference.md [hunk 5]
```diff
@@ -68,72 +71,189 @@ m.aepSdk = AdobeAEPSDKInit()
...
+```
+
+## Consent APIs:
+
+### setConsent
+
+Sends consent preferences to the Edge Network. For details on setting default consent before collecting user preferences, refer to the [Configure default consent](#configure-default-consent) section.
+
+> [!Important]
+> Please provide the entire payload with all the fields required by the Adobe 2.0 Standard.
```
Comment (praveek) [inline Documentation/api-reference.md:215]: Can you also add a link to the schema for Adobe 2.0 Standard?

Comment (addb) [reply Documentation/api-reference.md:215]: I did not find a proper link, but have linked to https://experienceleague.adobe.com/en/docs/experience-platform/landing/governance-privacy-security/consent/adobe/overview

Let me know if you know of a better document to link for this.

File: Documentation/api-reference.md [hunk 6]
```diff
@@ -68,72 +71,193 @@ m.aepSdk = AdobeAEPSDKInit()
...
+##### Example
+
+```brightscript
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+
+configuration = {}
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_CONFIG_ID] = "<YOUR_CONFIG_ID>"
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_DOMAIN] = "<YOUR_DOMAIN_NAME>"
+
+' Set the default consent to pending and update the consent later based on user preference
```
Comment (praveek) [inline Documentation/api-reference.md:169]: This example sets the default consent to pending. You can later update the consent based on user preferences using the setConsent API.

File: Documentation/api-reference.md [hunk 7]
```diff
@@ -68,72 +71,193 @@ m.aepSdk = AdobeAEPSDKInit()
...
+The default consent configuration determines how data collection consent is managed before invoking the [setConsent](#setconsent) API. This setting is particularly important to avoid unintentionally collecting data from users in regions where consent is required prior to data collection.
+
+By default, users are opted in for all purposes.
+
+#### Example: Setting default collect consent to pending before collecting actual user consent
+```brightscript
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+
+configuration = {}
+
```
Comment (praveek) [inline Documentation/api-reference.md:198]: This example sets the default consent to pending. You can later update the consent based on user preferences using the setConsent API.

**Review by praveek — APPROVED** [review]
Looks good. I just have a minor update to the comments.

**Pull request merged**